### ISSUE TITLE
JUnit5 compatibility?

### ISSUE THREAD

**Issue by pazz** [cs125-illinois/gradlegrader#4]
I have noticed that the demo test in your README is in JUnit4 syntax (`Test(timeout=300)` isn't possible in JUnit5). Have you tried gradlegrader with JUnit5 at all? I am running into a compilation error and suspect that one needs to tweak gradlegrader somehow.

[This is what I've tried](https://github.com/pazz/gradlegrader_template).
I can compile and test via gradle as expected. [My test class](https://github.com/pazz/gradlegrader_template/blob/master/src/test/java/uk/ac/liverpool/comp122/week1/TestHelloJava.java) contains a single unit test which is annotated using your `@Grade` decorator.

It seems that gradlegrader does not pick up on this test though:
```
 ./gradlew grade

> Task :relentlessCheckstyle
Checkstyle rule violations were found. See the report at: file:///home/pazz/repo/teaching/2021-COMP122/templates/junit5/build/reports/checkstyle/relentlessCheckstyle.xml
Checkstyle files with violations: 1
Checkstyle violations by severity: [warning:1]


> Task :test

TestHelloJava > testHelloWorld() PASSED

> Task :grade FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':grade'.
> java.lang.NoSuchMethodException: uk.ac.liverpool.comp122.week1.TestHelloJava.testHelloWorld()()

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 3s
8 actionable tasks: 6 executed, 2 up-to-date
```

I suspect that the issue is with the extra pair of brackets at the end of the test.
When setting up a gradle project with JUnit5 tests, it seems that the report files for tests are in a slightly different format. For example, [in my demo project](https://github.com/pazz/gradlegrader_template), `build/test-results/test/TEST-uk.ac.liverpool.comp122.week1.TestHelloJava.xml`
 contains 

```xml
<?xml version="1.0" encoding="UTF-8"?>
<testsuite name="uk.ac.liverpool.comp122.week1.TestHelloJava" tests="1" skipped="0" failures="0" errors="0" timestamp="2020-05-23T13:25:05" hostname="piu" time="0.029">
  <properties/>
  <testcase name="testHelloWorld()" classname="uk.ac.liverpool.comp122.week1.TestHelloJava" time="0.029"/>
  <system-out><![CDATA[]]></system-out>
  <system-err><![CDATA[]]></system-err>
</testsuite>
```
Notice the trailing braces in the testcase name `testHelloWorld()` . When using gradle with JUnit4, these are not present. I noticed this difference already when debugging my own ad-hoc automarking scripts this year.

The solution seems to be either to convince gradle somehow to stick with the old syntax, or adjust gradlegrader's code.

Any help would be much appreciated; Thanks again for this tool and the support :)

**Comment by Fleex255** [comment]
Thank you for the detailed example and investigation! Indeed, the extra name decoration from JUnit 5 broke GradleGrader's ability to find the test method. (We only used JUnit 4 because Robolectric requires it.) I just published version 2020.1.3 to fix this. If you're using JUnit 5 repeatable tests, each instance will be allocated the same number of points specified in `@Graded`.

**Comment by pazz** [comment]
wow, that was quick! I can confirm that it works as expected now. Thanks!
